Sources/Sentry/SentryTracerConcurrency.mm

```diff
@@ -1,49 +1,123 @@
 #import "SentryTracerConcurrency.h"
-#import "SentryId.h"
-#import "SentryLog.h"
-#include <mutex>
 
 #if SENTRY_TARGET_PROFILING_SUPPORTED
 
-static NSMutableSet<NSString *> *_gInFlightTraceIDs;
+#    import "SentryId.h"
+#    import "SentryLog.h"
+#    import "SentryProfiler+Private.h"
+#    import "SentryTracer.h"
+#    include <mutex>
+
+#    if SENTRY_HAS_UIKIT
+#        import "SentryDependencyContainer.h"
+#        import "SentryFramesTracker.h"
+#        import "SentryScreenFrames.h"
+#    endif // SENTRY_HAS_UIKIT
+
+// a mapping of profilers to the tracers that started them that are still in-flight and will need to
+// query them for their profiling data when they finish. this helps resolve the incongruity between
+// the different timeout durations between tracers (500s) and profilers (30s), where a transaction
+// may start a profiler that then times out, and then a new transaction starts a new profiler, and
+// we must keep the aborted one around until its associated transaction finishes.
+typedef NSMutableDictionary</* SentryProfiler.profileId */ NSString *,
+    NSMutableSet<SentryTracer *> *>
+    SentryProfilerTracerMapping;
+static SentryProfilerTracerMapping *_gProfilersToTracers;
+
+// provided for fast access to a profiler given a tracer
+typedef NSMutableDictionary</* SentryTracer.tracerId */ NSString *, SentryProfiler *>
+    SentryTracerProfilerMapping;
+static SentryTracerProfilerMapping *_gTracersToProfilers;
```

Comment:
nit: matter of opinion but I find this pattern of typedef'ing generic types makes code less readable at the site where the typedef is used, where it is not clear that it's just an alias for a specialized dictionary type. Even though repeating it is verbose, it is clearer to read.

Reply:
I agree with @indragiek.